types/reference/image_reference.go

```diff
@@ -0,0 +1,28 @@
+package reference
+
+import (
+	"fmt"
+
+	distreference "github.com/docker/distribution/reference"
+)
+
+// parseReference parses the given references and return the repository and
+// tag (if present) from it. If there is an error during parsing, it will
+// return an error.
+func Parse(ref string) (string, string, error) {
```

Comment:
@stevvooe yes. I think I'm gonna work on `docker/docker` with vendoring this branch to look at the result and see if I did miss something :angel: